Xamarin.Essentials/Calendar/Calendar.android.cs

```diff
@@ -1,7 +1,56 @@
-﻿namespace Xamarin.Essentials
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using Android.Provider;
+
+namespace Xamarin.Essentials
 {
     public static partial class Calendar
     {
-        public static bool IsSupported => true;
+        public static bool PlatformIsSupported => true;
+
+        public static async Task<IReadOnlyList<ICalendar>> PlatformGetCalendarsAsync()
+        {
+            await Permissions.RequireAsync(PermissionType.CalendarRead);
+
+            var calendarsUri = CalendarContract.Calendars.ContentUri;
+            var calendarsProjection = new List<string>
+            {
+                CalendarContract.Calendars.InterfaceConsts.Id,
+                CalendarContract.Calendars.InterfaceConsts.CalendarDisplayName,
+                CalendarContract.Calendars.InterfaceConsts.CalendarAccessLevel
+            };
+
+            var cur = Platform.AppContext.ApplicationContext.ContentResolver.Query(calendarsUri, calendarsProjection.ToArray(), null, null, null);
+            var calendars = new List<ICalendar>();
+            while (cur.MoveToNext())
+            {
+                calendars.Add(new DeviceCalendar()
+                {
+                    Id = cur.GetString(calendarsProjection.IndexOf(CalendarContract.Calendars.InterfaceConsts.Id)),
+                    Name = cur.GetString(calendarsProjection.IndexOf(CalendarContract.Calendars.InterfaceConsts.CalendarDisplayName)),
+                    IsReadOnly = IsCalendarReadOnly(cur.GetInt(calendarsProjection.IndexOf(CalendarContract.Calendars.InterfaceConsts.CalendarAccessLevel)))
```

Comment:
Parse to CalendarAccess enum and pass the value into the IsCalendarReadOnly method